tests/test_push.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from mozci.push import Push
+from mozci.task import Task
+
+
+# XXX: Tests to write:
+# doesn't run in first, passes in second, fails in third -> not regression of first
+# passes in first, fails in second, fails in third -> not regression of first
+# fails in first, passes in second, passes in third -> regression of first
+
+
+def test_didnt_run_in_current_failed_in_child_failed_in_grandchild():
+    '''
+    Tests the scenario where a task didn't run in a given push, but it
+    run and failed in its following pushes.
+    '''
+    current = Push("current")
+    child1 = Push("child1")
+    child2 = Push("child2")
+    last = Push("last")
+
+    current.child = child1
+    current.tasks = []
+
+    child1.child = child2
+    child1.tasks = [Task.create(id="1", label="test-prova", result="testfailed", classification="not classified")]
+
+    child2.child = last
+    child2.tasks = [Task.create(id="1", label="test-prova", result="testfailed", classification="not classified")]
+
+    last.child = last
+    last.tasks = []
+
+    assert current.get_candidate_regressions("label") == {"test-prova": 1}
+
+
+def test_didnt_run_in_current_failed_in_child_passed_in_grandchild():
+    '''
+    Tests the scenario where a task didn't run in a given push, but it
+    run and failed in a second push and then run and succeeded in a
+    third push.
+    '''
+    current = Push("current")
+    child1 = Push("child1")
+    child2 = Push("child2")
+    last = Push("last")
+
+    current.child = child1
+    current.tasks = []
+
+    child1.child = child2
+    child1.tasks = [Task.create(id="1", label="test-prova", result="testfailed", classification="not classified")]
+
+    child2.child = last
+    child2.tasks = [Task.create(id="1", label="test-prova", result="success")]
```

Comment:
Nice, stubbing out Push objects is much nicer than trying to mock ActiveData. Not sure why I thought it would be more complicated.

Eventually we might want to build some fixtures to make it easier to generate Push instances like this, but this is fine for now.